tests/setup.py

```diff
@@ -53,11 +53,17 @@ def close_test_view(view):
 
 class TextDocumentTestCase(DeferrableTestCase):
 
+    def __init__(self, view):
+        super().__init__(view)
+        self.test_file_path = test_file_path
+        self.client = None
+
     def setUp(self) -> None:
-        self.view = sublime.active_window().open_file(test_file_path)
+        self.view = sublime.active_window().open_file(self.test_file_path)
         self.view.settings().set("auto_complete_selector", "text.plain")
         self.wm = windows.lookup(self.view.window())
-        self.client = MockClient(async_response=sublime_delayer(100))
+        if self.client is None:
+            self.client = MockClient(async_response=sublime_delayer(100))
```

Comment:
Can you explain why test_file_path must be configurable and why the change in client assignment?

Reply:
The test saves the file. If the test fails, then other tests would use the content in that file.

As for the customizable MockClient, my initial thought was to make a test for the sync timeouts, but then I realized that we were dealing with mock transports anyway. I reverted that part.